code/modules/mob/living/basic/jungle/leaper/leaper_abilities.dm

```diff
@@ -0,0 +1,255 @@
+	summon_type = list(/mob/living/basic/frog/frog_suicide)
+	summon_radius = 2
+	summon_amount = 2
+	max_summons = 2
+
+/datum/action/cooldown/spell/conjure/limit_summons/create_suicide_toads/post_summon(atom/summoned_object, atom/cast_on)
+	. = ..()
+	var/mob/living/summoned_toad = summoned_object
+	summoned_toad.faction = owner.faction ///so they dont attack the leaper or the wizard master
+
+/obj/projectile/leaper
+	name = "leaper bubble"
+	icon_state = "leaper"
+	paralyze = 50
+	damage = 0
+	range = 7
+	hitsound = 'sound/effects/snap.ogg'
+	nondirectional_sprite = TRUE
+	impact_effect_type = /obj/effect/temp_visual/leaper_projectile_impact
+
+/obj/projectile/leaper/on_hit(atom/target, blocked = 0, pierce_hit)
+	..()
+	if (!isliving(target))
+		return
+	var/mob/living/bubbled = target
+	if(iscarbon(target))
+		bubbled.reagents.add_reagent(/datum/reagent/toxin/leaper_venom, 5)
+		return
+	if(isanimal(target))
+		var/mob/living/simple_animal/bubbled_animal = bubbled
+		bubbled_animal.adjustHealth(25)
+		return
+	if (isbasicmob(target))
+		bubbled.adjustBruteLoss(25)
+
+/obj/projectile/leaper/on_range()
+	new /obj/structure/leaper_bubble(get_turf(src))
+	return ..()
+
+/obj/effect/temp_visual/leaper_projectile_impact
+	name = "leaper bubble"
+	icon = 'icons/obj/weapons/guns/projectiles.dmi'
+	icon_state = "leaper_bubble_pop"
+	layer = ABOVE_ALL_MOB_LAYER
+	plane = GAME_PLANE_UPPER_FOV_HIDDEN
+	duration = 3
+
+/obj/effect/temp_visual/leaper_projectile_impact/Initialize(mapload)
+	. = ..()
+	new /obj/effect/decal/cleanable/leaper_sludge(get_turf(src))
+
+/obj/effect/temp_visual/blood_drop_rising
+	name = "leaper bubble"
+	icon = 'icons/obj/weapons/guns/projectiles.dmi'
+	icon_state = "leaper"
+	layer = ABOVE_ALL_MOB_LAYER
+	plane = GAME_PLANE_UPPER_FOV_HIDDEN
+	duration = 1 SECONDS
+
+/obj/effect/temp_visual/blood_drop_rising/Initialize(mapload)
+	. = ..()
+	animate(src, pixel_y = base_pixel_y + 150, time = duration)
+
+/obj/effect/temp_visual/blood_drop_falling
+	name = "leaper bubble"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "blood_effect_falling"
+	layer = ABOVE_ALL_MOB_LAYER
+	plane = GAME_PLANE_UPPER_FOV_HIDDEN
+	duration = 0.7 SECONDS
+	pixel_y = 60
+
+/obj/effect/temp_visual/blood_drop_falling/Initialize(mapload)
+	. = ..()
+	addtimer(CALLBACK(src, PROC_REF(create_blood_structure)), duration)
+	animate(src, pixel_y = 0, time = duration)
+
+/obj/effect/temp_visual/blood_drop_falling/proc/create_blood_structure()
+	playsound(src, 'sound/effects/snap.ogg', 50, TRUE)
+	new /obj/structure/leaper_bubble(get_turf(src))
+
+/obj/effect/temp_visual/shadow_telegraph
+	name = "shadow"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "shadow_telegraph"
+	duration = 1.5 SECONDS
+
+/obj/effect/decal/cleanable/leaper_sludge
+	name = "leaper sludge"
+	desc = "A small pool of sludge, containing trace amounts of leaper venom."
+	icon = 'icons/effects/tomatodecal.dmi'
+	icon_state = "tomato_floor1"
+
+/obj/structure/leaper_bubble
```

Comment:
There's so many discrete things in this one file that they could do with DMdocs purely to break up "here is a new thing and what it does"
Maybe move some of this stuff into their own files but there is some overlap in what effects and structures they create...